tests/manage/pv_services/pvc_clone/test_encrypted_rbd_pvc_clone.py

```diff
@@ -103,14 +115,18 @@ def setup(
             pv_obj = pvc_obj.backed_pv_obj
             vol_handle = pv_obj.get().get("spec").get("csi").get("volumeHandle")
             self.vol_handles.append(vol_handle)
-            if kms.is_key_present_in_path(
-                key=vol_handle, path=self.vault.vault_backend_path
-            ):
-                log.info(f"Vault: Found key for {pvc_obj.name}")
-            else:
-                raise ResourceNotFoundError(f"Vault: Key not found for {pvc_obj.name}")
 
-    def test_pvc_to_pvc_clone(self, kv_version, pod_factory):
+            if kms_provider == constants.VAULT_KMS_PROVIDER:
+                if kms.is_key_present_in_path(
+                    key=vol_handle, path=self.kms.vault_backend_path
+                ):
+                    log.info(f"Vault: Found key for {pvc_obj.name}")
+                else:
+                    raise ResourceNotFoundError(
+                        f"Vault: Key not found for {pvc_obj.name}"
+                    )
```

Comment:
Are there no validations required for HPCS in the backend?

Reply:
No validations are required in case of storage class encryption(`above test cases are part of this`), however we would need some validation in case of cluster wide encryption i.e., when we deploy ODF with HPCS, we would add some validation for keys.